logback-core/src/main/java/ch/qos/logback/core/AsyncAppenderBase.java

```diff
@@ -152,12 +153,16 @@ private boolean isQueueBelowDiscardingThreshold() {
   }
 
   private void put(E eventObject) {
-    try {
-      blockingQueue.put(eventObject);
-    } catch (InterruptedException e) {
-      // Interruption of current thread when in doAppend method should not be consumed
-      // by AsyncAppender
-      Thread.currentThread().interrupt();  	
+    if (neverBlock) {
+      blockingQueue.offer(eventObject);
```

Comment:
At this point, if you're using AsyncAppender, you're already silently dropping logs at INFO level and below. 

If you have a mechanism to safely notify that we dropped something, that couldn't hurt, but I don't think this patch makes things any worse. You've already explicitly said (in your config) you prefer dropping your ERROR logs instead of blocking your application.

I do vaguely recall I thought about setting a flag and constructing and emitting a notification log message if the queue cleaned out again, but testing the edge boundary seemed tricky, since you might be flapping on the edge of capacity and make the problem worse.